server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -484,11 +491,31 @@ public synchronized IndexShard createShard(
                     warmer.warm(reader, shard, IndexService.this.indexSettings);
                 }
             };
-
+            DiscoveryNode node = targetNode;
             Store remoteStore = null;
-            if (this.indexSettings.isRemoteStoreEnabled()) {
-                Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
-                remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+            boolean seedRemote = false;
+            if (node.isRemoteStoreNode()) {
+                if (this.indexSettings.isRemoteStoreEnabled()) {
+                    Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
+                    remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+                } else {
+                    logger.info("Hitting the migration case");
+                    if (sourceNode != null && sourceNode.isRemoteStoreNode() == false) {
+                        assert routing.primary();
```

Comment:
how are we ensuring that this is the primary relocation? I am guessing that we are handling that in relocation code. May be we can add more assertions here.

Reply:
If the index is not remote enabled, but the node is , it is migration case. This will hit for other cases than primary relocation as well .

For `(sourceNode != null && sourceNode.isRemoteStoreNode() == false)` , this will be hit only in relocation as we move primary to remote first , followed by replicas

Reply:
`routing.primary()` should always be true when source node is docrep and target node is remote . Because we always migrate primaries first to remote , followed by replica .